src/services/identity/LaunchClient.ts

```diff
@@ -35,21 +40,130 @@ class LaunchClient {
-  ): Promise<CreateDomainAssociationCommandOutput> =>
-    this.amplifyClient.send(new CreateDomainAssociationCommand(input))
+  ): Promise<CreateDomainAssociationCommandOutput> => {
+    if (this.isTestEnv()) {
+      return this.mockCreateDomainAssociationOutput(input)
+    }
+    const output = this.amplifyClient.send(
+      new CreateDomainAssociationCommand(input)
+    )
+    return output
+  }
 
   createGetDomainAssociationCommandInput = (
     appId: string,
-    domainName: string
-  ): GetDomainAssociationCommandInput => ({
+    domainName: string,
+    subDomainSettings: SubDomainSetting[]
+  ):
+    | GetDomainAssociationCommandInput
+    | MockGetDomainAssociationCommandInput => ({
     appId,
     domainName,
+    ...(this.isTestEnv() && { subDomainSettings }),
   })
 
   sendGetDomainAssociationCommand = (
-    input: GetDomainAssociationCommandInput
-  ): Promise<GetDomainAssociationCommandOutput> =>
-    this.amplifyClient.send(new GetDomainAssociationCommand(input))
+    input:
+      | GetDomainAssociationCommandInput
+      | MockGetDomainAssociationCommandInput
+  ): Promise<GetDomainAssociationCommandOutput> => {
+    const isMockInput = "subDomainSettings" in input
+    if (isMockInput) {
+      // handle mock input
+      return this.mockGetDomainAssociationOutput(input)
+    }
+    return this.amplifyClient.send(new GetDomainAssociationCommand(input))
+  }
+
+  /**
+   * The rate limit for Create Domain Association is 10 per hour.
+   * We want to limit interference with operations, as such we mock this call during development.
+   * @returns Mocked output for CreateDomainAssociationCommand
+   */
+  mockCreateDomainAssociationOutput = (
+    input: CreateDomainAssociationCommandInput
+  ): Promise<CreateDomainAssociationCommandOutput> => {
+    const mockResponse: CreateDomainAssociationCommandOutput = {
+      $metadata: {
+        httpStatusCode: 200,
+      },
+      domainAssociation: {
+        autoSubDomainCreationPatterns: [],
+        autoSubDomainIAMRole: undefined,
+        certificateVerificationDNSRecord: undefined,
+        domainAssociationArn: `arn:aws:amplify:ap-southeast-1:11111:apps/${input.appId}/domains/${input.domainName}`,
+        domainName: input.domainName,
+        domainStatus: "CREATING",
+        enableAutoSubDomain: false,
+        statusReason: undefined,
+        subDomains: undefined,
+      },
+    }
+
+    const subDomainSettingsList = input.subDomainSettings
+    if (!subDomainSettingsList || subDomainSettingsList.length === 0) {
+      return Promise.resolve(mockResponse)
+    }
+
+    const subDomains: SubDomain[] = this.getSubDomains(subDomainSettingsList)
+
+    // We know that domainAssociation is not undefined, so we can use the non-null assertion operator
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    mockResponse.domainAssociation!.subDomains = subDomains
+    return Promise.resolve(mockResponse)
+  }
+
+  private isTestEnv() {
```

Comment:
should we make this a generic util function?

Reply:
bc6aa75c already have util function 
 keeping the `isTestEnv()` function because I feel using that over `isSecure` represents the intent of the code better